custom/src/test/java/com/solarwinds/opentelemetry/extensions/initialize/config/HttpSettingsReaderTest.java

```diff
@@ -0,0 +1,224 @@
+    ConfigManager.reset();
+  }
+
+  @Test
+  void testGetSettings_UrlWithHttps() throws InvalidConfigException {
+    String collectorUrl = "https://apm.collector.na-01.cloud.solarwinds.com";
+    String serviceKey = "test-api-key:test-service";
+    String hostname = "test-hostname";
+
+    String expectedUrl =
+        "https://apm.collector.na-01.cloud.solarwinds.com/v1/settings/test-service/test-hostname";
+    String expectedTokenHeader = "Bearer test-api-key";
+    ConfigManager.setConfig(ConfigProperty.AGENT_COLLECTOR, collectorUrl);
+
+    ConfigManager.setConfig(ConfigProperty.AGENT_SERVICE_KEY, serviceKey);
+    try (MockedStatic<ResourceCustomizer> resourceCustomizerMock =
+        mockStatic(ResourceCustomizer.class)) {
+
+      Resource mockResource = mock(Resource.class);
+      resourceCustomizerMock.when(ResourceCustomizer::getResource).thenReturn(mockResource);
+      when(mockResource.getAttribute(HostIncubatingAttributes.HOST_NAME)).thenReturn(hostname);
+
+      when(delegate.fetchSettings(expectedUrl, expectedTokenHeader)).thenReturn(mockSettings);
+      Settings result = tested.getSettings();
+
+      assertNotNull(result);
+      assertEquals(mockSettings, result);
+      verify(delegate).fetchSettings(expectedUrl, expectedTokenHeader);
+    }
+  }
+
+  @Test
+  void testGetSettings_UrlWithoutHttps() throws InvalidConfigException {
+    String collectorUrl = "apm.collector.na-01.cloud.solarwinds.com";
+    String serviceKey = "test-api-key:test-service";
+    String hostname = "test-hostname";
+
+    String expectedUrl =
+        "https://apm.collector.na-01.cloud.solarwinds.com/v1/settings/test-service/test-hostname";
+    String expectedTokenHeader = "Bearer test-api-key";
+    ConfigManager.setConfig(ConfigProperty.AGENT_COLLECTOR, collectorUrl);
+
+    ConfigManager.setConfig(ConfigProperty.AGENT_SERVICE_KEY, serviceKey);
+    try (MockedStatic<ResourceCustomizer> resourceCustomizerMock =
+        mockStatic(ResourceCustomizer.class)) {
+      Resource mockResource = mock(Resource.class);
+      resourceCustomizerMock.when(ResourceCustomizer::getResource).thenReturn(mockResource);
+      when(mockResource.getAttribute(HostIncubatingAttributes.HOST_NAME)).thenReturn(hostname);
+
+      when(delegate.fetchSettings(expectedUrl, expectedTokenHeader)).thenReturn(mockSettings);
+      Settings result = tested.getSettings();
+
+      assertNotNull(result);
+      assertEquals(mockSettings, result);
+      verify(delegate).fetchSettings(expectedUrl, expectedTokenHeader);
+    }
+  }
+
+  @Test
+  void testGetSettings_UrlWithHttp() throws InvalidConfigException {
+    String collectorUrl = "http://apm.collector.na-01.cloud.solarwinds.com";
+    String serviceKey = "test-api-key:test-service";
+    String hostname = "test-hostname";
+
+    String expectedUrl =
+        "http://apm.collector.na-01.cloud.solarwinds.com/v1/settings/test-service/test-hostname";
+    String expectedTokenHeader = "Bearer test-api-key";
```

Comment:
Ooooh, this test is highlighting that we'd be sending the API token in cleartext if somehow an HTTP endpoint is set. Wonder if we should even allow this scenario?

Reply:
Duh, i missed the `s` in the expected URL! 👍